sh-runtime.c

```diff
@@ -829,7 +682,7 @@ DEPENDS_ON(pack_string)
   putstr("    : $((__cursor_fd$__fd = 0))         # Make buffer empty\n");
   putstr("    : $((__buflen_fd$__fd = 1000))      # Init buffer length\n");
   putstr("    : $((__state_fd$__fd = $3))         # Mark the fd as opened\n");
-  putstr("    pack_string $2\n");
+  putstr("    __res=$(_put_pstr __ $2)\n");
```

Comment:
This may be slower on certain shells, but opening a file is a relatively rare operation so it feels like a good compromise since it allows us to get rid of `pack_string`.